operator/api/redpanda/v1alpha2/common.go

```diff
@@ -236,11 +262,13 @@ type MetadataTemplate struct {
 type StaticConfigurationSource struct {
 	// Kafka is the configuration information for communicating with the Kafka
 	// API of a Redpanda cluster where the object should be created.
-	// +required
```

Comment:
because not all of the CRDs that use this use that part of the API spec (the schema CRD uses the schema registry APIs, not the Kafka API directly), I coupled the removal of `required` there with some additional validations where this is actually used and is required:

https://github.com/redpanda-data/redpanda-operator/blob/15a5ac8d60176d4455891ad06f646add4442a218/operator/api/redpanda/v1alpha2/topic_types.go#L47

and

https://github.com/redpanda-data/redpanda-operator/blob/15a5ac8d60176d4455891ad06f646add4442a218/operator/api/redpanda/v1alpha2/user_types.go#L91

Reply:
thx